### ISSUE TITLE
AWXRestore - Unmanaged Postgres - Override Postgres Secret

### ISSUE THREAD

**Issue by coolhome** [ansible/tower-operator#1726]
### Please confirm the following

- [X] I agree to follow this project's [code of conduct](https://docs.ansible.com/ansible/latest/community/code_of_conduct.html).
- [X] I have checked the [current issues](https://github.com/ansible/awx-operator/issues) for duplicates.
- [X] I understand that AWX Operator is open source software provided for free and that I might not receive a timely response.

### Feature Summary

When using AWXRestore it might be useful for unmanaged postgres to use a new database server, database name, etc. Without touching the backup storage maybe there is a way to overwrite the postgres configuration secret.

**Comment by fosterseth** [comment]
can you elaborate on how this feature would look?

**Comment by coolhome** [comment]
@fosterseth I was thinking something simple like adding this spec property to AWXRestore as optional. It is expected that the secret specified should be created before deploying the AWXRestore CR.

https://github.com/ansible/awx-operator/blob/fc11db4ececf3fa7197b3906f6a9c50e2e1b6bb1/config/crd/bases/awx.ansible.com_awxs.yaml#L68-L70

When recreating secrets and AWXRestore.spec.postgres_configuration_secret is set, omit secret from backup from being restored.

https://github.com/ansible/awx-operator/blob/fc11db4ececf3fa7197b3906f6a9c50e2e1b6bb1/roles/restore/tasks/secrets.yml#L81-L88

When deploying AWX CR, override AWX.spec.postgres_configuration_secret with AWXRestore.spec.postgres_configuration_secret.

https://github.com/ansible/awx-operator/blob/fc11db4ececf3fa7197b3906f6a9c50e2e1b6bb1/roles/restore/tasks/deploy_awx.yml#L3-L12

I haven't tested the following but it might be a way we can do this:

```yaml
- name: Omit Postgres Configuration Secret
  set_fact:
    secrets: "{{ secrets | dict2items | difference( secrets.postgresConfigurationSecret | dict2items ) | items2dict }}"
  when:
    - postgres_configuration_secret | length > 0
    - secrets.postgresConfigurationSecret is defined

- name: Override Postgres Configuration Secret
  set_fact:
    specs: "{{ specs | dict2items | union([{ 'key': 'postgres_configuration_secret', 'value': postgres_configuration_secret }]) | items2dict }}"
  when: postgres_configuration_secret | length > 0
```

**Comment by coolhome** [comment]
I believe this is solved with #1862
